src/psyclone/transformations.py

```diff
@@ -563,6 +566,394 @@ def apply(self, node, options=None):
+                                                            (Schedule, Loop)):
+                        node_vars.merge(VariablesAccessInfo(child))
+            else:
+                # If we find node types that could be our parents (are are
+                # known to not be valid options from previous checks) then
+                # we set node_vars to empty
+                node_vars = VariablesAccessInfo()
+            node_signatures = node_vars.all_signatures
+            # Once we have the node's variable accesses, check for collisions
+            for sig1 in taskloop_signatures:
+                # If this signature is not in the node signatures then continue
+                if sig1 not in node_signatures:
+                    continue
+                access1 = taskloop_vars[sig1]
+                access2 = node_vars[sig1]
+                # If both read only, we can ignore this signature
+                # Otherwise, one of themwrites so return this node as
+                # we have a WaR or RaW dependency
+                if access1.is_written() or access2.is_written():
+                    # If we have a different parent serial node, and
+                    # parent_single is not None then our parent_single is our
+                    # dependency, otherwise this node is the dependency
+                    if (taskloop.ancestor(OMPSerialDirective) is not
+                            node.ancestor(OMPSerialDirective) and
+                            parent_single is not None):
+                        return parent_single
+                    return node
+        # If we found no dependencies, then return that!
+        return None
+
+    def apply(self, node, options=None):
+        '''
+        Apply an OMPTaskwait Transformation to the supplied node
+        (which must be a OMPParallelDirective). In the generated code this
+        corresponds to adding zero or more OMPTaskwaitDirectives as
+        appropriate:
+
+        .. code-block:: fortran
+
+          !$OMP PARALLEL
+            ...
+            !$OMP TASKWAIT
+            ...
+            !$OMP TASKWAIT
+            ...
+          !$OMP END PARALLEL
+
+        :param node: the node to which to apply the transformation.
+        :type node: :py:class:`psyclone.f2pygen.DoGen`
+        :param options: a dictionary with options for transformations\
+                        and validation.
+        :type options: dictionary of string:values or None
+        :param bool options["fail_on_no_taskloop"]:
+                indicating whether this should throw an error if no \
+                OMPTaskloop nodes are found in this tree.
+
+        :returns: two-tuple of transformed schedule and a record of the \
+                  transformation.
+        :rtype: (:py:class:`psyclone.psyir.nodes.Schedule, \
+                 :py:class:`psyclone.undoredo.Memento`)
+        '''
+
+        self.validate(node, options=options)
+
+        schedule = node.root
```

Comment:
I'm not sure you want to do this or is it just here for the Memento?

Reply:
It was for momento, removed.